### PR TITLE
chalk-laugh: refactor Text component

### PR DESCRIPTION
**Pull request by taravancil**
## Links
* [chalk-laugh.glitch.me](https://chalk-laugh.glitch.me)
* [Manuscript](https://glitch.fogbugz.com/f/cases/3328263/Text-component-margins-and-font-size-adjustments)

<div style="display: flex">
  
  

</div>

## Changes:
* Add `size`, `weight`, `defaultMargin` props to Text component
* Allow Text component to accept `className`
* Update all instances of Text component to take advantage of these updates

## How To Test:
Try out a bunch of places where the Text component is used:
- Project count on [collection page](https://chalk-laugh.glitch.me/@taravancil/glitch)
- [Error page](https://chalk-laugh.glitch.me/404)
- [Home page "of course this site was made on glitch too"](https://chalk-laugh.glitch.me)

## Feedback I'm looking for:
- Instead of the `defaultMargin` prop, would we prefer `marginTop` and `marginBottom` props? This would eliminate some cases where we're using a `className` just to specify a margin
- As far as I can tell, there's nowhere the `.p + .p` selector is applied. Do we want to eliminate that and just use margin props? ^

### CODE REVIEW

File: src/components/text/text.styl [hunk 1]
```diff
@@ -2,4 +2,11 @@
 
 .p
   font-family: sansserif
-  line-height: 20px
\ No newline at end of file
+  font-size: var(--size);
+  font-weight: var(--weight);
+  line-height: 20px
```
Comment (modernserf) [inline src/components/text/text.styl]: should line-height be proportional to --size?

File: src/components/text/text.styl [hunk 2]
```diff
@@ -2,4 +2,11 @@
...
+  margin: 0
+  + .p
+    margin-bottom: 1em
```
Comment (modernserf) [inline src/components/text/text.styl]: i think you want `margin-top` here, since it applies to the _second_ `p`

Comment (taravancil) [reply src/components/text/text.styl]: Whoops, good catch

File: src/components/text/text.styl (hunk 1 above)
Comment (taravancil) [reply src/components/text/text.styl]: Ah, yep. Will update later after this workshop

**Pull request closed without merging**